### PR TITLE
Added an elementary unit test for CSP

### PR DESCRIPTION
**Pull request by kavyasrinet**
@abhinavarora and I designed this unit test for the proposed imperative programming paradigm in https://github.com/PaddlePaddle/Paddle/blob/develop/doc/design/csp.md .
This is a unit test that implements the Daisy Chain as explained here: https://talks.golang.org/2012/concurrency.slide#39 using the concepts of concurrent programming in Fluid.

### CODE REVIEW

File: python/paddle/v2/fluid/tests/notest_csp.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import paddle.v2.fluid as fluid
+
+
+class TestCSPFramework(unittest.TestCase):
+    def helper(self, left, right):
```
Comment (wangkuiyi) [inline python/paddle/v2/fluid/tests/notest_csp.py]: This function has only one line; why make it a function?

File: python/paddle/v2/fluid/tests/notest_csp.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import paddle.v2.fluid as fluid
+
+
+class TestCSPFramework(unittest.TestCase):
```
Comment (wangkuiyi) [inline python/paddle/v2/fluid/tests/notest_csp.py]: Could we make sure that this unit test doesn't break the CI before we finish the implementation of CSP?

Comment (kavyasrinet) [reply python/paddle/v2/fluid/tests/notest_csp.py]: Sure, will do.

File: python/paddle/v2/fluid/tests/notest_csp.py (hunk 1 above)
Comment (wangkuiyi) [reply python/paddle/v2/fluid/tests/notest_csp.py]: I see. I am sure that here we should not need to define the helper function.

File: python/paddle/v2/fluid/tests/notest_csp.py (hunk 2 above)
Comment (kavyasrinet) [reply python/paddle/v2/fluid/tests/notest_csp.py]: @wangkuiyi I made sure that the CI doesn't break. Both TeamCity and Travis pass.

**Pull request merged**